### PR TITLE
[FSDP2] Fixed 2D clip grad norm test

### PR DESCRIPTION
**Pull request by awgu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #126497

This fixes https://github.com/pytorch/pytorch/issues/126484.

We change from transformer to MLP stack since transformer seems to introduce slight numeric differences when using TP. We include a sequence parallel layer norm module in the MLP stack to exercise `(S(0), R)` placement.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

**Review by weifengpy — APPROVED** [review]

**Review by weifengpy — APPROVED** [review]

**Comment by weifengpy** [comment]
switching to fsdp2 release note is really some labor work

**Review by wz337 — APPROVED** [review]
LGTM! Thanks for looking into it so quickly.

**Comment by awgu** [comment]
Failure are all inductor-related, not FSDP2-related.

**Comment by awgu** [comment]
@jeanschmidt I wonder if we can re-land this. I do not think it is related to Inductor CUDA 12 failures.

File: .ci/pytorch/test.sh [hunk 1]
```diff
@@ -326,6 +326,7 @@ test_inductor_distributed() {
   python test/run_test.py -i distributed/_composable/fsdp/test_fully_shard_frozen.py --verbose
   python test/run_test.py -i distributed/_composable/fsdp/test_fully_shard_mixed_precision.py -k test_compute_dtype --verbose
   python test/run_test.py -i distributed/_composable/fsdp/test_fully_shard_mixed_precision.py -k test_reduce_dtype --verbose
+  python test/run_test.py -i distributed/_composable/fsdp/test_fully_shard_clip_grad_norm_.py -k test_clip_grad_norm_2d --verbose
```
Comment (awgu) [inline .ci/pytorch/test.sh:329]: Test name was incorrect before (fixed to include trailing `_`).

**Pull request closed without merging**